lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
-        # @return [Boolean] true if this adapter can handle this type of identifier
+        # @return [Boolean] true if this adapteppr can handle this type of identifier
         def handles?(id)
           id =~ /^10\.[A-Za-z0-9]+\/[A-Za-z0-9\-]+/
         end
 
-        # Generates a new persistent id, ensuring that nothing currently uses that identifier.
-        # @return [String] a new id
-        def mint
-          # metadata is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          metadata = {}
-          # target is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          target_url = nil
-          # status is reserved when minting
-          # TODO: allow provided target
-          identifier_status = IDENTIFIER_STATUS[:reserved]
-          # mint_identifier returns a Hyacinth::Ezid::Doi
-          ezid_doi_instance = api_session.mint_identifier(DATACITE[:shoulder][:doi], identifier_status, target_url, metadata)
-          # if above method returned nil, call to EZID API was unsuccessful. Return nil to indicate failure
-          # if unsuccessful because of timeout, log faiilure and return nil
-          if api_session.timed_out?
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier timed-out.")
-            return nil
+        # metadata is not required when minting a draft DOI.
+        # metadata is required when minting a findable DOI. This metadata must include all the
+        # properties required by DataCite.
+        # It is also possible to specify a (non-existent) DOI when minting. However, this is not
+        # currently supported in the code.
+        def mint(hyacinth_metadata = nil, # Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::Metadata
+                 target_url = nil, doi_state = :draft, _doi = nil)
+          if doi_state != :draft && hyacinth_metadata.nil?
+            Rails.logger.error "Hyacinth metadata required to mint DOI in #{doi_state} state"
+            return
           end
-          # if got a response from EZID server indicating failure, log info and return nil
-          if ezid_doi_instance.nil?
-            response = api_session.last_response_from_server
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier was unsuccessful.")
-            log(:error, "#mint_and_store_doi: Response from EZID server follows:\n#{http_error_message(response)}")
-            return nil
+          datacite_data = Datacite::RestApi::V2::Data.new
+          datacite_data.prefix = DATACITE[:prefix]
+          map_metadata(hyacinth_metadata, datacite_data, target_url) if hyacinth_metadata
+          datacite_data.build_mint(doi_state, hyacinth_metadata.present?)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          rest_api_response = rest_api.post_dois datacite_data.generate_json_payload
```

Comment:
```suggestion
          rest_api_response = rest_api.post_dois(datacite_data.generate_json_payload)
```
Minor formatting suggestion here since you call this method with parens in all other places (and I think it's easier to read).

Reply:
Minor formatting change has been applied.